nengo_bones/scripts/generate_bones.py

```diff
@@ -0,0 +1,94 @@
+"""Scripts for auto-generating nengo-bones files."""
+
+import os
+
+import click
+import jinja2
+
+import nengo_bones
+
+
+@click.group()
+@click.option("--conf-file", default=None, help="Filepath for config file")
+@click.option("--output-dir", default=".", help="Output directory for scripts")
+@click.option("--template-dir", default=None,
+              help="Directory containing additional templates")
+@click.pass_context
+def main(ctx, conf_file, output_dir, template_dir):
+    """Loads config file and sets up template environment."""
+
+    ctx.ensure_object(dict)
+
+    config = nengo_bones.load_config(conf_file)
+
+    template_dirs = [] if template_dir is None else [template_dir]
+    template_dirs.append(
+        os.path.join(os.path.dirname(__file__), "..", "templates"))
+    env = jinja2.Environment(
+        loader=jinja2.FileSystemLoader(template_dirs),
+        trim_blocks=True, lstrip_blocks=True, keep_trailing_newline=True)
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    ctx.obj["env"] = env
+    ctx.obj["config"] = config
+    ctx.obj["output_dir"] = output_dir
+
+
+@main.command()
+@click.pass_context
+def ci_scripts(ctx):
+    """Generate TravisCI shell scripts."""
+
+    config = ctx.obj["config"]
+    for params in config["script_config"]:
+        script_name = params.pop("template")
+        template = ctx.obj["env"].get_template(script_name + ".sh.template")
```

Comment:
What happens if there are two templates with the same name? e.g., if the repo defines its own `.ci/test.sh.template` that extends the nengo-bones `test.sh.template`?

Reply:
The local one (found in `--template-dir`) takes precedence.